parastell.py

```diff
@@ -202,15 +204,80 @@ def native_export():
 
     dir = Path(export['dir'])
     filename = Path(export['h5m_filename'])
+    
+    def merge_layer_surfaces():
+        #the plasma will have IDs 1,2,3 and name in components dict 'plasma'
+        #the ends are 'planar surfaces'
+        #the swept surfaces are 'spline surfaces'
+        #when using cubit.get_surface_type(int)
+        
+        # get a tuple of volume ids, these will be in the import order
+        # and are assumed to increase in order from inside to outside
+        volumes = cubit.get_entities('volume')
+
+        # tracks the surface id of the outer surface of the previous layer
+        lastOuterSurface = None
+        
+        for name, vol in zip(components.keys(),volumes):
+
+            # wait to merge until the next layer if the plasma is included
+            # store surface to be merged for next loop
+            if name == 'plasma':
+                surfaces = cubit.get_relatives('volume',vol,'surface')
+
+                for surf in surfaces:
+                    if cubit.get_surface_type(surf) == 'spline surface':
+                        lastOuterSurface = surf
+
+            # check if we are in the first layer in a build with plasma excluded
+            # store outer surface to be merged in next loop
+            elif lastOuterSurface is None:
+                surfaces = cubit.get_relatives('volume',vol,'surface')
+                splineSurfaces = []
+
+                for surf in surfaces:
+                    if cubit.get_surface_type(surf) == 'spline surface':
+                        splineSurfaces.append(surf)
+
+                if cubit.get_bounding_box('surface', splineSurfaces[1])[4] > cubit.get_bounding_box('surface',splineSurfaces[0])[4]:
+                    outerSurface = splineSurfaces[1]
+
+                else:
+                    outerSurface = splineSurfaces[0]
+
```

Comment:
Maybe this could be a function `orient_component_spoine_surfaces(vol)` that returns `[innerSurface, outerSurface]` and then use it here and in the next conditional?

Reply:
Or you do this operation once in both cases, and then act on it differently depending on which case.

Reply:
added local `orient_spline_surfaces` function to improve readability